src/components/GrampsjsMap.js

```diff
@@ -67,66 +70,94 @@ class GrampsjsMap extends LitElement {
     this.mapid = 'mapid'
     this.latitude = 0
     this.longitude = 0
+    this.year = -1
     this.latMin = 0
     this.latMax = 0
     this.longMin = 0
     this.longMax = 0
     this.layerSwitcher = false
+    this._currentLayer = 'OpenStreetMap'
   }
 
   firstUpdated() {
     const mapel = this.shadowRoot.getElementById(this.mapid)
     if (this.latMin === 0 && this.latMax === 0) {
-      this._map = new Map(mapel, {zoomControl: false}).setView(
+      this._map = new L.Map(mapel, {zoomControl: false}).setView(
         [this.latitude, this.longitude],
         this.zoom
       )
     } else {
-      this._map = new Map(mapel, {zoomControl: false}).fitBounds([
+      this._map = new L.Map(mapel, {zoomControl: false}).fitBounds([
         [this.latMin, this.longMin],
         [this.latMax, this.longMax],
       ])
     }
     const config = {...defaultConfig, ...window.grampsjsConfig}
-    const tileLayer = new TileLayer(config.leafletTileUrl, {
+    const tileLayer = new L.TileLayer(config.leafletTileUrl, {
       attribution: config.leafletTileAttribution,
       tileSize: config.leafletTileSize,
       zoomOffset: config.leafletZoomOffset,
       maxZoom: 19,
       zoomControl: false,
     })
     tileLayer.addTo(this._map)
-    this._map.addControl(control.zoom({position: 'bottomright'}))
+    this._gl = L.maplibreGL({
+      style: config.glStyle,
+    })
+    this._map.addControl(L.control.zoom({position: 'bottomright'}))
     if (this.locateControl) {
       this._map.addControl(
-        control.locate({position: 'bottomright', drawCircle: false})
+        L.control.locate({position: 'bottomright', drawCircle: false})
       )
     }
-    this._layercontrol = control.layers({OpenStreetMap: tileLayer}, null, {
-      position: 'bottomleft',
-    })
+    this._layercontrol = L.control.layers(
+      {
+        OpenHistoricalMap: this._gl,
+        OpenStreetMap: tileLayer,
+      },
+      null,
+      {
+        position: 'bottomleft',
+      }
+    )
     if (this.layerSwitcher) {
       this._map.addControl(this._layercontrol)
     }
+    this._map.on('baselayerchange', e => this._handleBaseLayerChange(e))
     this._map.invalidateSize(false)
     this._map.on('moveend', e => this._handleMoveEnd(e))
   }
 
+  _handleBaseLayerChange(e) {
+    this._currentLayer = e.name
+    fireEvent(this, 'map:layerchange', {layer: this._currentLayer})
+  }
+
   _handleMoveEnd(e) {
     fireEvent(this, 'map:moveend', {bounds: e.target.getBounds()})
   }
 
   panTo(latitude, longitude) {
     if (this._map !== undefined) {
-      this._map.panTo(new LatLng(latitude, longitude))
+      // eslint-disable-next-line new-cap
+      this._map.panTo(new L.latLng(latitude, longitude))
     }
   }
 
-  updated() {
+  updated(changed) {
+    if (
+      changed.has('year') &&
+      this.year > 0 &&
+      this._currentLayer === 'OpenHistoricalMap'
+    ) {
+      filterByDecimalYear(this._gl._glMap, this.year)
```

Comment:
`this._gl.getMaplibreMap()` is probably more robust than accessing this internal property.